terraso_backend/apps/shared_data/visualization_tileset_tasks.py

```diff
@@ -0,0 +1,161 @@
+        )
+
+
+def remove_mapbox_tileset(tileset_id):
+    if tileset_id is None:
+        return
+    try:
+        remove_tileset(tileset_id)
+    except Exception as error:
+        logger.exception(
+            "Error deleting mapbox tileset",
+            extra={"tileset_id": tileset_id, "error": str(error)},
+        )
+
+
+def create_mapbox_tileset(visualization_id):
+    logger.info("Creating mapbox tileset", visualization_id=visualization_id)
+    visualization = VisualizationConfig.objects.get(pk=visualization_id)
+    data_entry = visualization.data_entry
+    group_entry = visualization.group
+
+    # Delete mapbox tileset if needed
+    remove_mapbox_tileset(visualization.mapbox_tileset_id)
+
+    try:
+        rows = get_rows_from_file(data_entry)
+
+        first_row = rows[0]
+
+        dataset_config = visualization.configuration["datasetConfig"]
+        annotate_config = visualization.configuration["annotateConfig"]
+
+        longitude_column = dataset_config["longitude"]
+        longitude_index = first_row.index(longitude_column)
+
+        latitude_column = dataset_config["latitude"]
+        latitude_index = first_row.index(latitude_column)
+
+        data_points = annotate_config["dataPoints"]
+        data_points_indexes = [
+            {
+                "label": data_point.get("label", data_point["column"]),
+                "index": first_row.index(data_point["column"]),
+            }
+            for data_point in data_points
+        ]
+
+        annotation_title = annotate_config.get("annotationTitle")
+
+        title_index = (
+            first_row.index(annotation_title)
+            if annotation_title and annotation_title in first_row
+            else None
+        )
+
+        features = []
+        for row in rows:
+            fields = [
+                {
+                    "label": data_point["label"],
+                    "value": row[data_point["index"]],
+                }
+                for data_point in data_points_indexes
+            ]
+
+            properties = {
+                "title": row[title_index] if title_index else None,
+                "fields": json.dumps(fields),
+            }
+
+            try:
+                longitude = float(row[longitude_index])
+                latitude = float(row[latitude_index])
+                feature = {
+                    "type": "Feature",
+                    "geometry": {
+                        "type": "Point",
+                        "coordinates": [longitude, latitude],
+                    },
+                    "properties": properties,
+                }
+
+                features.append(feature)
+            except ValueError:
+                continue
+
+        geojson = {
+            "type": "FeatureCollection",
+            "features": features,
+        }
+        logger.info(
+            "Geojson generated for mapbox tileset",
+            visualization_id=visualization_id,
+            rows=len(rows),
+        )
```

Comment:
question: does this need to be in production? should it be logger.debug?

Reply:
Same answer as before